internal/bminventory/inventory.go

```diff
@@ -3798,14 +3798,20 @@ func (b *bareMetalInventory) DownloadMinimalInitrd(ctx context.Context, params i
 	var netFiles []staticnetworkconfig.StaticNetworkConfigData
 	var scriptContent, serviceContent string
 	if infraEnv.StaticNetworkConfig != "" {
+		var includesMACIdentifier bool
+		includesMACIdentifier, err = b.staticNetworkConfig.CheckConfigForMACIdentifier(infraEnv.StaticNetworkConfig)
+		if err != nil {
+			return common.GenerateErrorResponder(err)
+		}
+
 		// backward compatibility - nmstate.service has been available on RHCOS since version 4.14+, therefore, we should maintain both flows
 		var ok bool
```

Comment:
For clarity, worth renaming the 'ok' var to something like 'isNmstateServiceSupported'.